pkg/cloudprovider/types.go

```diff
@@ -215,10 +215,11 @@ func (i InstanceTypeOverhead) Total() v1.ResourceList {
 }
 
 // An Offering describes where an InstanceType is available to be used, with the expectation that its properties
-// may be tightly coupled (e.g. the availability of an instance type in some zone is scoped to a capacity type)
+// may be tightly coupled (e.g. the availability of an instance type in some zone is scoped to a capacity type) and
+// these properties are captured with labels in the Constraints map. v1beta1.CapacityTypeLabelKey and
+// v1.LabelTopologyZone are required to be set.
 type Offering struct {
-	CapacityType string
-	Zone         string
+	Requirements scheduling.Requirements
```

Comment:
Does it make sense to put the invariants that are enforced by these Offerings in terms of requirement labels over this Requirements block here?

Reply:
Something like `Requirement is required to contain the keys..."